pkg/types/validation/installconfig.go

```diff
@@ -542,9 +542,22 @@ func validateNamedRepository(r string) error {
 		// in an error. Instead we will check whether the input is
 		// a valid hostname as a workaround.
 		if err == dockerref.ErrNameNotCanonical {
-			_, err := url.ParseRequestURI(r)
-			if err != nil {
-				return fmt.Errorf("the repository provided is invalid")
+			// First check whether the repo name provided contains
+			// host:port. If so, split the host part and lookup
+			// to see if it is a valid host. If the port is
+			// not provided, then just check the name to see if
+			// the repo name resolves correctly.
+			host, _, err := net.SplitHostPort(r)
+			if err == nil {
+				_, err = net.LookupHost(host)
```

Comment:
A valid registry might only be accessible to the bootstrap node but not the installer host. In which case this would incorrectly fail validation.

Reply:
You're right. If we can't resolve the name then I'll keep it simple and look for a valid dns name string characters.